src/fetchers.ts

```diff
@@ -282,6 +297,48 @@ async function fetchHelm(
   });
 }
 
+async function fetchNixpkgs(
+  spec: NixpkgsFetch,
+  context: FetchContext,
+): Promise<Version[]> {
+  const nixexprs = await fetchUrl("https://channels.nixos.org/" + spec.channel, context, undefined, { redirect: "manual" }, async (response) => {
+    if (response.status !== 301) {
+      throw new Error(
+        `Got status ${response.status}: ${await response.text()}`,
+      );
+    }
+    const location = response.headers.get("Location");
+    if (location === null) {
+      throw new Error(`Could not get location for ${response.url}`);
+    }
+    return location + "/nixexprs.tar.xz";
+  });
+  let hash = undefined as string | undefined;
+  if (!context.update) {
+    const cached = context.cache.getBody(nixexprs);
+    if (cached) {
+      hash = cached;
+    }
+  }
+  if (hash === undefined) {
+    const p = Deno.run({
+      cmd: [
+        "nix-prefetch-url",
```

Comment:
This is a hash of the contents of the tarball, so as a "version" I don't know if this is the most useful thing to return. I think for nixpkgs that's pinned to a channel like this, it'd be easier to understand (and less work) to use the channel url. In the table of differences, it's easier to compare `23.05pre467690.0cd51a933d9` and `23.05pre462724.8e3fad82be6`.

For projects with pinned nixpkgs that don't use our `generate-pin.rb` or don't use channel urls, it's probably better to use git commits.

Reply:
point taken